### DISCUSSION TITLE
Multi-domain SSR Caching

### DISCUSSION THREAD

**Discussion by skene** [vercel/vercel#5234] [General] [answered]
We routed several domains (foo.com, bar.com) to a single Vercel project in order to accomplish dynamic SSR based on the requested domain. We've got it working via Next.js getServerSideProps, although caching the serverless lambda page responses returns to most recently requested domain, and not the active requested domain; understandably as the cache is at the deployment level and not domain. Is there a caching configuration that makes domain level caching possible? 

**Possible Workaround**
Use CloudFlare to cache at the domain level, and purge on each deployment; although this is not ideal, it makes it possible.

**Similar Inquiries** 
https://github.com/vercel/vercel/discussions/4092
https://github.com/vercel/vercel/discussions/3950

Thanks in advanced.

**Comment by styfle** [comment, marked as answer]
Hi @skene 

It sounds like you want to use a different cache based on the hostname of the request, even though the underlying deployment is the same.

This is not possible currently, but we are tracking internally as Story 3703.

Today, you would probably need to create two different projects, each with their own domain.

Thanks!

**Reply by hacktisch** [reply]
@styfle Is there a way to subscribe to updates on "internal Story 3703"?
I am in the same situation as skene

**Reply by styfle** [reply]
This feature has been rolled out and your deployments will use a different cache based on the hostname.

**Reply by WorldLanguages** [reply]
@styfle Hi, I'm in the opposite position as the main post, I'd like to share the cache between all domains. Is there a way to return to the previous behavior?

**Comment by styfle** [comment]
@WorldLanguages Can you share the use case for a shared cache between domains?

**Reply by WorldLanguages** [reply]
I DMed you through Twitter with the specific description of my issue so you can confirm whether it's related to this change or not - it likely is but I'm not 100% sure, so I'd rather have someone confirm before going off-topic.
